src/bin/sccache-dist/cmdline/parse.rs

```diff
@@ -0,0 +1,340 @@
+    fn from(log_level: LogLevel) -> Self {
+        match log_level {
+            LogLevel::Error => Self::Error,
+            LogLevel::Warn => Self::Warn,
+            LogLevel::Info => Self::Info,
+            LogLevel::Debug => Self::Debug,
+            LogLevel::Trace => Self::Trace,
+        }
+    }
+}
+
+fn flag_infer_long(name: &'static str) -> Arg<'static> {
+    Arg::new(name).long(name)
+}
+
+fn get_clap_command() -> ClapCommand<'static> {
+    let syslog = flag_infer_long("syslog")
+        .help("Log to the syslog with LEVEL")
+        .value_name("LEVEL")
+        .possible_values(LogLevel::values().iter().map(LogLevel::as_str));
+    let config_with_help_message =
+        |help: &'static str| flag_infer_long("config").help(help).value_name("PATH");
+
+    ClapCommand::new(env!("CARGO_PKG_NAME"))
+        .version(env!("CARGO_PKG_VERSION"))
+        .subcommand_required(true)
+        .subcommand(
+            ClapCommand::new("auth")
+                .subcommand_required(true)
+                .subcommand(ClapCommand::new("generate-jwt-hs256-key"))
+                .subcommand(
+                    ClapCommand::new("generate-jwt-hs256-server-token")
+                        .args(&[
+                            flag_infer_long("server")
+                                .help("Generate a key for the specified server")
+                                .value_name("SERVER_ADDR")
+                                .required(true),
+                            flag_infer_long("secret-key")
+                                .help("Use specified key to create the token")
+                                .value_name("KEY"),
+                            config_with_help_message(
+                                "Use the key from the scheduler config file at PATH",
+                            ),
+                        ])
+                        .group(
+                            ArgGroup::new("key_source_mutual_exclusion")
+                                .args(&["config", "secret-key"])
+                                .required(true),
+                        ),
+                )
+                .subcommand(
+                    ClapCommand::new("generate-shared-token").arg(
+                        flag_infer_long("bits")
+                            .help("Use the specified number of bits of randomness")
+                            .value_name("BITS")
+                            .default_value("256")
+                            .validator(TokenLength::from_bits),
+                    ),
+                ),
+        )
+        .subcommand(ClapCommand::new("scheduler").args(&[
+            config_with_help_message("Use the scheduler config file at PATH").required(true),
+            syslog.clone(),
+        ]))
+        .subcommand(ClapCommand::new("server").args(&[
+            config_with_help_message("Use the server config file at PATH").required(true),
+            syslog,
+        ]))
```

Comment:
Is there a particular reason you chose the manual construction vs creating rather than using the proc-macro provided by clap?

Reply:
If you are talking about the clap's derive API, then the PR was originally using that, but switched to the builder API in 13ae495ab76b0826eadac0a7e9f87caedcfbeb0c in response to this review comment https://github.com/mozilla/sccache/pull/1140#pullrequestreview-910722058

I personally prefer the derive API since you no longer have to make sure that both parsing and consuming match, but the concerns around less flexibility and worse documentation are certainly valid

Reply:
I see, I missed that one. Thank you! Leave as is, I wasn't aware of the discussion.